module/os/freebsd/zfs/arc_os.c

```diff
@@ -211,6 +210,7 @@ arc_prune_async(int64_t adjust)
 
 	*adjustptr = adjust;
 	taskq_dispatch(arc_prune_taskq, arc_prune_task, adjustptr, TQ_SLEEP);
+	taskq_wait(arc_prune_taskq);
```

Comment:
Now that this synchronous, have you done any performance testing? From what I've seen in the past, this can take a long time to traverse all the filesystems and all of the znodes.